packages/next/src/client/script.tsx

```diff
@@ -57,7 +57,9 @@ const loadScript = (props: ScriptProps): void => {
     LoadCache.add(cacheKey)
     // It is possible that multiple `next/script` components all have same "src", but has different "onLoad"
     // This is to make sure the same remote script will only load once, but "onLoad" are executed in order
-    ScriptCache.get(src).then(onLoad, onError)
+    ScriptCache.get(src).then((result: unknown) => {
+      return result instanceof Event ? onError?.(result) : onLoad?.(result)
+    })
```

Comment:
IMHO, you should not use `Event` to distinguish `onError` from `onLoad` (and that's exactly why the integration test case failed).

Reply:
Thanks, I'm on it! 

I think I've got a test for this scenario working and have also managed to fix the integration tests. Hopefully I can push the changes soon enough.

Reply:
Alright so the solution was to re-throw.

```ts
const throws = new Promise((resolve, reject) => {
  reject(new Error('Always reject'))
}).catch((e) => {
  console.log('rejected')
  throw e
})

async function main() {
  const promise = throws

  promise.then(
    () => console.log('left'),
    () => console.log('right')
  )
}

main()
```

This one would print `"right"` in case of error. Applied the same to the error branch, and added a test for this situation